diff --git a/config/pathdef.c.in b/config/pathdef.c.in
index 4579fbe..97c37bb 100644
--- a/config/pathdef.c.in
+++ b/config/pathdef.c.in
@@ -1,5 +1,5 @@
 #include "${PROJECT_SOURCE_DIR}/src/nvim/vim.h"
 char *default_vim_dir = "${CMAKE_INSTALL_FULL_DATAROOTDIR}/nvim";
-char *default_vimruntime_dir = "";
+char *default_vimruntime_dir = "$MINGW_PREFIX/share/nvim/runtime";
 char_u *compiled_user = (char_u *)"${USERNAME}";
 char_u *compiled_sys = (char_u *)"${HOSTNAME}";
