diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 7562045..e10e73f 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -54,7 +54,7 @@ if(NOT APPLE)
 			DESTINATION ${CMAKE_INSTALL_DATADIR}/pixmaps/)
 endif()
 
-if(WIN32 AND NOT CMAKE_CROSSCOMPILING)
+if(WIN32 AND NOT CMAKE_CROSSCOMPILING AND NOT MINGW)
 	if(Qt5Core_VERSION VERSION_LESS 5.3)
 		message(WARNING "Cannot use windeployqt in Qt 5.2 and earlier")
 	else()
