diff --git a/src/luarocks/cfg.lua b/src/luarocks/cfg.lua
index 48acd93..b8f2f98 100644
--- a/src/luarocks/cfg.lua
+++ b/src/luarocks/cfg.lua
@@ -416,14 +416,6 @@ if cfg.platforms.windows then
    defaults.variables.CFLAGS = "/nologo /MD /O2"
    defaults.variables.LIBFLAG = "/nologo /dll"
 
-   local bins = { "SEVENZ", "CP", "FIND", "LS", "MD5SUM",
-      "MKDIR", "MV", "PWD", "RMDIR", "TEST", "UNAME", "WGET" }
-   for _, var in ipairs(bins) do
-      if defaults.variables[var] then
-         defaults.variables[var] = full_prefix.."\\tools\\"..defaults.variables[var]
-      end
-   end
-
    defaults.external_deps_patterns = {
       bin = { "?.exe", "?.bat" },
       lib = { "?.lib", "?.dll", "lib?.dll" },
@@ -451,11 +443,11 @@ end
 
 if cfg.platforms.mingw32 then
    defaults.obj_extension = "o"
-   defaults.cmake_generator = "MinGW Makefiles"
-   defaults.variables.MAKE = "mingw32-make"
-   defaults.variables.CC = "mingw32-gcc"
+   defaults.cmake_generator = "MSYS Makefiles"
+   defaults.variables.MAKE = "make"
+   defaults.variables.CC = "gcc"
    defaults.variables.RC = "windres"
-   defaults.variables.LD = "mingw32-gcc"
+   defaults.variables.LD = "gcc"
    defaults.variables.CFLAGS = "-O2"
    defaults.variables.LIBFLAG = "-shared"
    defaults.external_deps_patterns = {
